src/handlers/events-query.ts

```diff
@@ -0,0 +1,43 @@
+import type { DidResolver } from '../index.js';
+import type { EventLog } from '../types/event-log.js';
+import type { MethodHandler } from '../types/method-handler.js';
+import type { EventsQueryMessage, EventsQueryReply } from '../types/event-types.js';
+
+import { EventsQuery } from '../interfaces/events-query.js';
+import { messageReplyFromError } from '../core/message-reply.js';
+import { authenticate, authorize } from '../core/auth.js';
+
+
+export class EventsQueryHandler implements MethodHandler {
+
+  constructor(private didResolver: DidResolver, private eventLog: EventLog) { }
+
+  public async handle({
+    tenant,
+    message
+  }: {tenant: string, message: EventsQueryMessage}): Promise<EventsQueryReply> {
+    let eventsQuery: EventsQuery;
+
+    try {
+      eventsQuery = await EventsQuery.parse(message);
+    } catch (e) {
+      return messageReplyFromError(e, 400);
+    }
+
+    try {
+      await authenticate(message.authorization, this.didResolver);
+      await authorize(tenant, eventsQuery);
```

Comment:
I realized I never updated the error message in `authorize()`. Though it's tangential to this PR, do you mind updating that error message to something other than the current "permissions check not yet implemented"? Maybe, "Message failed authorization. Only the tenant is authorized."